adapters/Oath/OathAdapter/GADMVASAdapterBaseClass.m

```diff
@@ -0,0 +1,365 @@
+{
+    return [self.connector viewControllerForPresentingModalView];
+}
+
+- (void)inlineAdDidRefresh:(nonnull VASInlineAdView *)inlineAd
+{
+    // Unused
+}
+
+- (void)inlineAdDidResize:(nonnull VASInlineAdView *)inlineAd
+{
+    // Unused
+}
+
+- (void)inlineAdEvent:(nonnull VASInlineAdView *)inlineAd source:(nonnull NSString *)source eventId:(nonnull NSString *)eventId arguments:(nonnull NSDictionary<NSString *,id> *)arguments
+{
+    // Unused
+}
+
+#pragma mark - common
+
+- (void)stopBeingDelegate
+{
+    if (self.inlineAd) {
+        if ([self.inlineAd respondsToSelector:@selector(destroy)]) {
+            [self.inlineAd performSelector:@selector(destroy)];
+        } else {
+            NSLog(@"GADMAdapterVerizon: The adapter is intended to work with Verizon Ads SDK version 1.0.1 or higher.  Please update the Verizon Ads SDK.");
+        }
+    }
+    
+    if (self.interstitialAd) {
+        if ([self.interstitialAd respondsToSelector:@selector(destroy)]) {
+            [self.interstitialAd performSelector:@selector(destroy)];
+        } else {
+            NSLog(@"GADMAdapterVerizon: The adapter is intended to work with Verizon Ads SDK version 1.0.1 or higher.  Please update the Verizon Ads SDK.");
+        }
+    }
+    
+    self.inlineAdFactory.delegate = nil;
+    self.inlineAd.delegate = nil;
+    self.interstitialAdFactory.delegate = nil;
+    self.interstitialAd.delegate = nil;
+    self.inlineAd = nil;
+    self.interstitialAd = nil;
+}
+
+
+#pragma mark - private
+
+- (void)setRequestInfoFromConnector:(id<GADMediationAdRequest>)connector
+{
+    //User Settings
+    [self setUserSettingsFromConnector:connector];
+    
+    //COPPA
+    [self setCoppaFromConnector:connector];
+    
+    // Location
+    if (connector.userHasLocation) {
+        self.vasAds.locationEnabled = YES;
+    }
+}
+
+- (void)setUserSettingsFromConnector:(id<GADMediationAdRequest>)connector
+{
+    VASRequestMetadataBuilder *builder = [[VASRequestMetadataBuilder alloc] init];
+    
+    // Mediator
+    builder.appMediator = [NSString stringWithFormat:@"AdMobVAS-%@",  [GADMVASAdapterBaseClass adapterVersion]];
+    
+    // Keywords
+    if ([connector userKeywords] != nil && [[connector userKeywords] count] > 0) {
+        builder.userKeywords = [connector userKeywords];;
+    }
+    
+    self.vasAds.requestMetadata = [builder build];
+}
+
+- (void)setCoppaFromConnector:(id<GADMediationAdRequest>)connector
+{
+    self.vasAds.COPPA = [connector childDirectedTreatment];
+}
+
+- (CGRect)makeBannerRect:(GADAdSize)adSize
+{
+    // Consideration for smart banners ( http://bit.ly/1wAWn0r )
+    CGSize windowSize = [[UIApplication sharedApplication] keyWindow].bounds.size;
+    
+    if ( GADAdSizeEqualToSize(kGADAdSizeSmartBannerPortrait, adSize) ||
+        GADAdSizeEqualToSize(kGADAdSizeSmartBannerLandscape, adSize) ) {
+        int offsetX = (int)(windowSize.width / 2);
+        
+        if ( windowSize.height < MM_SMARTBANNER_MED_CUTOFF ) {
```

Comment:
Should these be checking for width instead of height?

When making a SmartBanner request for a phone, it's trying to return a size applicable for a tablet (MM_SMARTBANNER_TABLET) instead of for a phone, causing the returning banner to be too large for the ad view.